k8s/pkg/apis/longhorn/v1beta1/engine.go

```diff
@@ -11,105 +11,175 @@ const (
-	Labels      map[string]string `json:"labels"`
+	// +optional
+	Name string `json:"name"`
+	// +optional
+	Parent string `json:"parent"`
+	// +optional
+	Children map[string]bool `json:"children"`
+	// +optional
+	Removed bool `json:"removed"`
+	// +optional
+	UserCreated bool `json:"usercreated"`
+	// +optional
+	Created string `json:"created"`
+	// +optional
+	Size string `json:"size"`
+	// +optional
+	Labels map[string]string `json:"labels"`
 }
 
 type EngineSpec struct {
-	InstanceSpec
-	Frontend                  VolumeFrontend    `json:"frontend"`
-	ReplicaAddressMap         map[string]string `json:"replicaAddressMap"`
+	InstanceSpec `json:""`
+
+	// +optional
+	Frontend VolumeFrontend `json:"frontend"`
+	// +optional
+	ReplicaAddressMap map[string]string `json:"replicaAddressMap"`
+	// +optional
 	UpgradedReplicaAddressMap map[string]string `json:"upgradedReplicaAddressMap"`
-	BackupVolume              string            `json:"backupVolume"`
-	RequestedBackupRestore    string            `json:"requestedBackupRestore"`
-	RequestedDataSource       VolumeDataSource  `json:"requestedDataSource"`
-	DisableFrontend           bool              `json:"disableFrontend"`
-	RevisionCounterDisabled   bool              `json:"revisionCounterDisabled"`
+	// +optional
+	BackupVolume string `json:"backupVolume"`
+	// +optional
+	RequestedBackupRestore string `json:"requestedBackupRestore"`
+	// +optional
+	RequestedDataSource VolumeDataSource `json:"requestedDataSource"`
+	// +optional
+	DisableFrontend bool `json:"disableFrontend"`
+	// +optional
+	RevisionCounterDisabled bool `json:"revisionCounterDisabled"`
 }
 
 type EngineStatus struct {
-	InstanceStatus
-	CurrentSize              int64                           `json:"currentSize,string"`
-	CurrentReplicaAddressMap map[string]string               `json:"currentReplicaAddressMap"`
-	ReplicaModeMap           map[string]ReplicaMode          `json:"replicaModeMap"`
-	Endpoint                 string                          `json:"endpoint"`
-	LastRestoredBackup       string                          `json:"lastRestoredBackup"`
-	BackupStatus             map[string]*BackupStatus        `json:"backupStatus"`
-	RestoreStatus            map[string]*RestoreStatus       `json:"restoreStatus"`
-	PurgeStatus              map[string]*PurgeStatus         `json:"purgeStatus"`
-	RebuildStatus            map[string]*RebuildStatus       `json:"rebuildStatus"`
-	CloneStatus              map[string]*SnapshotCloneStatus `json:"cloneStatus"`
-	Snapshots                map[string]*Snapshot            `json:"snapshots"`
-	SnapshotsError           string                          `json:"snapshotsError"`
-	IsExpanding              bool                            `json:"isExpanding"`
-	LastExpansionError       string                          `json:"lastExpansionError"`
-	LastExpansionFailedAt    string                          `json:"lastExpansionFailedAt"`
+	InstanceStatus `json:""`
+
+	// +kubebuilder:validation:Type=string
```

Comment:
It's weird to use string as a validation, why not `int64`. If needed to change, probably we need to change the json tag?

Reply:
Shuo has the same question before.
It's because we store string in the ETCD.

We could change the data type to int64 in Longhorn v1beta2 version, and I had created an issue for this already.